tests/test_cli.py

```diff
@@ -254,3 +256,54 @@ def test_revoke_token(default_user, session):
         ],
     )
     assert "does not have an active access token" in result.output
+
+
+class TestMessageConsumer:
+    def test_do_not_remove_message(
+        self,
+        in_memory_queue_connection,
+        default_in_memory_producer,
+        consume_queue,
+    ):
+        """Test if MessageConsumer ignores and re-queues not matching message."""
+        workflow_name = "workflow.1"
+        queue_name = "workflow-submission"
+        consumer = MessageConsumer(
+            connection=in_memory_queue_connection,
+            queue_name=queue_name,
+            message_key="workflow_id_or_name",
+            messages_to_delete=["some_other_name"],
+        )
+        in_memory_wsp = WorkflowSubmissionPublisher(
+            connection=in_memory_queue_connection
+        )
+
+        in_memory_wsp.publish_workflow_submission("1", workflow_name, {})
+        consume_queue(consumer, limit=1)
+        assert not in_memory_queue_connection.channel().queues[queue_name].empty()
+        in_memory_queue_connection.channel().queues.clear()
```

Comment:
*issue (non-blocking):* if in-memory queues are not cleared tests fail

The test fails for both MessageConsumer and scheduler. This is probably due to [in_memory_queue_connection having a "session" scope](https://github.com/reanahub/pytest-reana/blob/0055f5c72d55bdabe0e5c68bb714f5f177f95f18/pytest_reana/fixtures.py#L626). It means some messages can stay in the queue between tests and cause unexpected failures.

As for now, I solved it by cleaning messages at the end of each test. But, this is not a sustainable solution. It is not nice when you need to remember to clear stuff after the test. I propose to create an issue to address later.

WDYT?

Reply:
Yes, I think that opening an issue is a good idea!

Reply:
It can be a nice "good first issue"